server/api/repos/[repo_id]/chat.post.ts

```diff
@@ -1,26 +1,40 @@
-import { Octokit } from 'octokit';
+import { repoSchema, userReposSchema } from "../../../schemas";
+import { eq } from 'drizzle-orm';
+
 
 export default defineEventHandler(async (event) => {
   const config = useRuntimeConfig();
-  const token = getHeader(event, 'gh_token');
-  const octokit = new Octokit({ auth: token });
-
-  // TODO: check user access to repo
 
   const repoId = event.context.params?.repo_id;
+
   if (!repoId) {
     throw createError({
       statusCode: 400,
       statusMessage: 'repo_id is required',
     });
   }
 
+  const repo = await db.select().from(repoSchema).where(eq(repoSchema.id, Number(repoId))).get();
+
+  const user = await getUserFromCookie(event);
```

Comment:
```suggestion
  const user = await getUserFromCookie(event);
  if (!user) {
     return sendError(
      event,
      createError({
        statusCode: 401,
        message: 'Unauthorized',
      }),
    );
  }
```

So you don't need the if nesting later.